lib/util/filesystem.ml

```diff
@@ -103,19 +103,20 @@ let walk_folder ?(recursive = false) f path acc =
   in
   traverse [ path ] acc
 
-let copy_file ?(perm = 0o640) src dst =
+let copy_file ?(perm = 0o644) src dst =
   let sz = 8192 in
   let buf = Bytes.create sz in
-  Compat.In_channel.with_open_bin src @@ fun ic ->
-  Compat.Out_channel.with_open_gen
-    [ Open_wronly; Open_append; Open_trunc ]
-    perm dst
-  @@ fun oc ->
-  let rec loop () =
+  let rec loop ic oc =
     match Compat.In_channel.input ic buf 0 sz with
     | 0 -> ()
     | r ->
         Compat.Out_channel.output oc buf 0 r;
-        loop ()
+        loop ic oc
   in
-  loop ()
+  Compat.In_channel.with_open_bin src @@ fun ic ->
+  if Sys.win32 then Compat.Out_channel.with_open_bin dst @@ fun oc -> loop ic oc
```

Comment:
The permission is ignored on Windows after this modification. Did you try to add `Open_binary` in the flags of `with_open_gen`? This could fix the bug. I thought that binary mode was the default.